src/docs/automation-and-resiliency/horizontal-and-vertical-autoscaling-demo.md

```diff
@@ -0,0 +1,1011 @@
+    $ oc delete vpa example-vpa-recommender
+    verticalpodautoscaler.autoscaling.k8s.io "example-vpa-recommender" deleted
+
+    $ oc get vpa
+    No resources found in tats-vpa-test namespace.
+
+    // Delete one of pod
+    $ oc delete pod example-74979bff7b-hdh92 --grace-period=0 --force
+    warning: Immediate deletion does not wait for confirmation that the running resource has been terminated. The resource may continue to run on the cluster indefinitely.
+    pod "example-74979bff7b-hdh92" force deleted
+ 
+    // Pod will be recreated with the original CPU and Memory requests by the deployment:
+
+    NAME                       READY   STATUS    RESTARTS   AGE   IP             NODE                  NOMINATED NODE   READINESS GATES
+    example-74979bff7b-9bt4n   1/1     Running   0          69s   10.97.18.109   mcs-klab-app-04.dmz   <none>           <none>
+    example-74979bff7b-qp4cs   1/1     Running   0          12m   10.97.17.193   mcs-klab-app-02.dmz   <none>           <none>
+    NAME                        VPA_UPDATES
+    example-74979bff7b-9bt4n   <none>
+    example-74979bff7b-qp4cs   Pod resources updated by example-vpa-recommender: container 0: memory request, cpu request, cpu limit, memory limit
+    NAME                        CPU_REQ    MEM_REQ    CPU_LIM    MEM_LIM
+    example-74979bff7b-9bt4n   5m         10Mi       100m       100Mi
+    example-74979bff7b-qp4cs   25m        262144k    500m       2500Mi
+    ```
+
+    Note: The VPA does not change the Deployment settings. Therefore, when the VPA is deleted, the Pod will revert to its original required values.
+
+## HPA + VPA for easy resource tuning demo
+
+The best practices in our OpenShift are the following 7 steps;
+
+- Use VPA to learn optimum resource settings for your deployments:
+
+    Step 1: Identify deployments in your namespace that you think need a resource increase/decrease
+
+    Step 2: Create a VPA object for these deployments
+
+    Step 3: Let the VPA object(s) run for 2-7 days to collect data
+  
+    Step 4:  Review the report with the recommendations for resource allocations generated by the VPA objects
+
+    Step 5: Implement for resource allocations for CPU Request/Limit and Memory Request/Limit recommended by the VPA objects
+
+- Use HPA to make your deployments scalable:
+
+    Step 6: Create a HPA object for the deployment that is expected to burst and set the threshold for scaling out to the CPU Limit set for the deployment
+
+    Step 7: Run a load test to generate a load that will trigger the HPA and increase the number of replicas to confirm everything works as expected
+
+### Use VPA to learn optimum resource settings for your deployments
+
+#### Step 1
+
+Identify deployments in your namespace that you think need a resource increase/decrease
+
+Follow the steps in [Example Nginx Application Deployment](ExampleAppDeployment.md) and deploy a demo app in your namespace. In this Demo, `tats-autoscaling-demo` namespace is used.
```

Comment:
This will cause a broken link. We don't have any example Nginx app deployment documentation to point to. Not too sure what to do here.

Reply:
Note for my future self: we're looking at just intergration the ExampleAppDeployment.md into horizontal-and-vertical-autoscaling-demo.md